utils/build/virtual_machine/provisions/auto-inject/auto-inject_installer_manual.yml

```diff
@@ -48,8 +48,29 @@
     export $(cat scenario_agent.env | xargs)
 
     sudo -E sh -c "sudo mkdir -p /etc/datadog-agent && printf \"api_key: ${DD_API_KEY}\nsite: datadoghq.com\n\" > /etc/datadog-agent/datadog.yaml"
-    DD_REPO_URL=${DD_injection_repo_url} bash -c "$(curl -L https://install.datadoghq.com/scripts/install_script_agent7.sh)"
     
+    
+    # dns install.datadoghq.com or the network interfaces are no very steady, so we need to retry
+    # On the old machines the curl command doesn't support the --retry-connrefused flag, we implement the retry policy in a loop
+    # standard exec: DD_REPO_URL=${DD_injection_repo_url} bash -c "$(curl -L https://install.datadoghq.com/scripts/install_script_agent7.sh)"
+    MAX_RETRIES=5
+    RETRY_DELAY=5
+    COUNTER=0
+
+    while [ $COUNTER -lt $MAX_RETRIES ]; do
+        curl -L https://install.datadoghq.com/scripts/install_script_agent7.sh -o install_script.sh && break
```

Comment:
If the version of curl is recent enough, there is a built-in retry mechanism: https://everything.curl.dev/usingcurl/downloads/retry.html